kernel/kernel-impl/src/main/java/org/sakaiproject/authz/impl/SakaiSecurity.java

```diff
@@ -1094,8 +1106,17 @@ protected User newMockupStudentInSite(Site site, String eid, String role) throws
 		return newUser;
 	}
 	
+	// Instance variables to store original user's session information
+	private String originalUserId;
+	private String originalUserEid;
```

Comment:
These class member variables are visible globally and should not store data that is session scoped.

Reply:
It hadn't occurred to me before, but you're right. I'll make sure to address this in my upcoming commit for the role swap feature. Thanks for pointing it out, Earle.